windirstat/WinDirStat.cpp

```diff
@@ -284,6 +284,15 @@ BOOL CDirStatApp::InitInstance()
     COptions::LoadAppSettings();
     LoadStdProfileSettings(0);
 
+    // Silently restart elevated conditionally before any expensive initialization
+    if (IsElevationAvailable() && COptions::ElevateOnDemand && !COptions::ShowElevationPrompt // only if user doesn't want to be prompted
+        // Any of these options enabled in settings would trigger elevation
+        && (COptions::UseFastScanEngine || COptions::UseBackupRestore))
```

Comment:
I think I mentioned this before, but there are other reasons to run elevated even if you don't have these available so I would remove them from the logic.  If the user can elevate and they have opted to before, just do it.